### PR TITLE
75/create user context & create requests page & request card

### LINKED ISSUE #31: Create requests card
As a **user** I want to **be able to see a booking card** so i can **easily see information about my sent and recieved requests**

Each card should contain rentee name, rentee picture, message, requested item for rent and requested dates for rental.

**Acceptance criteria**

Sent booking card should include: 
- [ ] Username/name of lessor

- [ ] Item title
- [ ] Price 
- [ ] Answer from lessor (Accepted/declined/pedning)

_If we have time_ 
- [ ] Lessors score
- [ ] Picture of item requested for rent
- [ ] Time from/to
- [ ] Message to lessor

Received booking card should include: 
- [ ] Username/name of renter
- [ ] Item title
- [ ] Price 
- [ ] Button for accepting or declining the request

_If we have time_
- [ ] Renters score
- [ ] Picture of item requested for rent
- [ ] Time from/to
- [ ] Message from renter

### LINKED ISSUE #33: Create requests page
As a **user** I want to **have a reqests page** so i can **see my sent and received requests**

**Acceptance criteria**
_- [ ] Notification badge for new requests_ OLD! New issue #74 
- [ ] Requests displayed as [booking cards](https://github.com/mirrenil/arbetsmetodik/issues/31)
- [x] Only accessible when logged in

_Nice to have_
- [ ] Button for toggling old requests

### LINKED ISSUE #75: Create user context
As a developer I want all functionality for fetching and storing data related to current user to be in a user context

### PR DESCRIPTION
**Pull request by liinneea-a**
closes #75 
closes #33 
closes #31 

This PR will: 
Update contexts. 
Add a userContext which handles all data related to the user while authContext only handles all related to authentication.
Update name on interface for list item.
create a request card:  ( _data for sent from is set to email. This will change when displayName is set on all users_ )

create a requests page

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -10,9 +10,10 @@ import SignInPage from "./Pages/SignIn";
 import SignUpPage from "./Pages/SignUp";
 import StartPage from "./Pages/StartPage";
 import { createTheme, colors, ThemeProvider, Typography } from "@mui/material";
-import AuthProvider from "./Contexts/AuthContext";
+import { AuthProvider } from "./Contexts/AuthContext";
```
Comment (mirrenil) [inline src/App.tsx:13]: Why the curly braces here and not around UserProvider?

File: src/Components/Header.tsx [hunk 2]
```diff
@@ -209,6 +234,12 @@ const Header = () => {
             <Link to="/howItWorks" style={{ textDecoration: "none" }}>
               <Typography sx={itemsDesk}>How it works</Typography>
             </Link>
+            <Link to="/" style={{ textDecoration: "none" }}>
+              <Typography sx={itemsDesk}>My requests</Typography>
+            </Link>
+            <Link to="/signin" style={{ textDecoration: "none" }}>
```
Comment (mirrenil) [inline src/Components/Header.tsx:240]: Remove the link to singin/signup

**Review by mirrenil — CHANGES_REQUESTED** [review]
Big one! Nice work!! Left some comments to fix before me merge 💯 🥇

File: src/App.tsx (hunk 1 above)
Comment (liinneea-a) [reply src/App.tsx:13]: I blame prettier for that :3

File: src/Components/Header.tsx (hunk 2 above)
Comment (liinneea-a) [reply src/Components/Header.tsx]: Done!

**Review by mirrenil — APPROVED** [review]
👯

**Pull request merged**